src/wireviz/wireviz.py

```diff
@@ -86,6 +86,9 @@ def parse(
         raise Exception("No output formats or return types specified")
 
     yaml_data, yaml_file = _get_yaml_data_and_path(inp)
+    assert isinstance(
+        yaml_data, dict
+    ), f"Expected a dict/map as top-level YAML input, but got: {type(yaml_data)}"
```

Comment:
In runtime code, it is better to handle type checks this way. `assert` statements are still applicable to have in software tests.

```suggestion
    if not isinstance(yaml_data, dict):
        raise TypeError(f"Expected a dict/map as top-level YAML input, but got: {type(yaml_data)}"
```

Reply:
Thanks!